pan-domain-auth-verification/src/main/scala/com/gu/pandomainauth/service/Crypto.scala

```diff
@@ -25,19 +24,25 @@ object Crypto {
 
   def signData(data: Array[Byte], prvKey: PrivateKey): Array[Byte] = {
     val rsa = Signature.getInstance(signatureAlgorithm, "BC")
-    val privateKey = keyFactory.generatePrivate(new PKCS8EncodedKeySpec(decodeBase64(prvKey.key)))
-    rsa.initSign(privateKey)
+    rsa.initSign(prvKey)
 
     rsa.update(data)
     rsa.sign()
   }
 
   def verifySignature(data: Array[Byte], signature: Array[Byte], pubKey: PublicKey) : Boolean = {
     val rsa = Signature.getInstance(signatureAlgorithm, "BC")
-    val publicKey = keyFactory.generatePublic(new X509EncodedKeySpec(decodeBase64(pubKey.key)))
-    rsa.initVerify(publicKey)
+    rsa.initVerify(pubKey)
```

Comment:
The `signData()` &`verifySignature()` methods on `Crypto` no-longer need to repeatedly decode the base64-encoded strings, and so now are a bit simpler and more efficient.